### PR TITLE
mqtt: limit maximum number of live transactions

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/5430

Describe changes:
- mqtt: limit maximum number of live transactions

### CODE REVIEW

**Comment by catenacyber** [comment]
Another option would be to prevent a transaction to be pushed into the transactions list/Vec
But that would require changing the 31 occurrences of `self.transactions.push`

Another option would be to free directly the transactions instead of marking them as complete, but they would not get logged...

This is the best I have come up with... Not sure if it is good enough for 60x

**Comment by jasonish** [comment]
> Another option would be to free directly the transactions instead of marking them as complete, but they would not get logged...

This is what we used to do with DNS, and I think its a fair emergency strategy along with a flooded alert which you have.

File: rust/src/mqtt/mqtt.rs [hunk 1]
```diff
@@ -52,6 +54,7 @@ pub enum MQTTEvent {
     MissingMsgId,
     UnassignedMsgtype,
     MalformedTraffic,
+    FloodedFlushed,
```
Comment (jasonish) [inline rust/src/mqtt/mqtt.rs:57]: Should be `TooManyTransactions` as thats what we already have in master with more or less the same meaning.

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | TREX_GENERIC_stats_chk | | |
| .capture.kernel_drops | 37050 | 89591 | 241.81% |
| .ftp.memuse | 0 | 56 | - | 

Pipeline 9437

**Comment by catenacyber** [comment]
> > Another option would be to free directly the transactions instead of marking them as complete, but they would not get logged...
> 
> This is what we used to do with DNS, and I think its a fair emergency strategy along with a flooded alert which you have.

You do not have the flooded alert, if you free the alert, because you do not have time to run detection...

Closing in favor of Jeff's alternative MR

**Pull request closed without merging**